### ISSUE TITLE
Multiple conflicting spotless configurations resolved in same project

### ISSUE THREAD

**Issue by anuraaga** [diffplug/spotless#1034]
If you are submitting a **bug**, please include the following:

- [ ] summary of problem
- [ ] gradle or maven version
- [ ] spotless version
- [ ] operating system and version
- [ ] copy-paste your full Spotless configuration block(s), and a link to a public git repo that reproduces the problem if possible
- [ ] copy-paste the full content of any console errors emitted by `gradlew spotless[Apply/Check] --stacktrace`

If you're just submitting a feature request or question, no need for the above.

With Gradle 7.3.1, spotless 6.0.4, I get this error on my build. It seems to be sporadic behavior. TBH these are making me wish for the old days of the tool resolved in the buildscript 😅 

```
* What went wrong:
Could not determine the dependencies of task ':instrumentation:opentelemetry-api:opentelemetry-api-1.4:javaagent:byteBuddyJava'.
> Could not resolve all task dependencies for configuration ':instrumentation:opentelemetry-api:opentelemetry-api-1.4:javaagent:runtimeClasspath'.
   > Could not resolve project :instrumentation:opentelemetry-api:opentelemetry-api-1.0:javaagent.
     Required by:
         project :instrumentation:opentelemetry-api:opentelemetry-api-1.4:javaagent
      > The consumer was configured to find a runtime of a library, packaged as a jar, preferably optimized for standard JVMs, and its dependencies declared externally. However we cannot choose between the following variants of project :instrumentation:opentelemetry-api:opentelemetry-api-1.0:javaagent:
          - spotless-1972456287
          - spotless865452459
        All of them match the consumer attributes:
          - Variant 'spotless-1972456287' capability io.opentelemetry.opentelemetry-api-1.0:javaagent:1.10.0-alpha-SNAPSHOT declares a component, and its dependencies declared externally:
              - Unmatched attributes:
                  - Doesn't say anything about its component category (required a library)
                  - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)

              - Other compatible attributes:
                  - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
                  - Doesn't say anything about its elements (required them packaged as a jar)
```

Repo (should hit this commit since we'll need to revert for now)
https://github.com/open-telemetry/opentelemetry-java-instrumentation/commit/4ae6d1a57c699ee82b4750a6a0ca7d19e658a61e

I get the failure with the command `./gradlew :instrumentation:opentelemetry-api:opentelemetry-api-1.0:javaagent:check`

**Comment by anuraaga** [comment]
BTW in case it helps, I have written plugins before that provision tools - the pattern I found to work best was to have a separate setup plugin applied to `project.getRootPlugin()`. Gradle ensures it's only applied once, so the project plugin can do it in the code, users don't worry about it. Without registering it with the plugin plugin, it is just a class that implements `Plugin` and has minimal leakage into user projects. Just a random suggestion

**Comment by nedtwigg** [comment]
Weird! #984 is the tracking issue for declaring and resolving deps in the root project, agreed that is a good idea. Would be even better if Spotless could "just work"...

**Comment by nedtwigg** [comment]
Should be fixed in `6.1.0`, please reopen if the issue persists.  See [these docs](https://github.com/diffplug/spotless/tree/main/plugin-gradle#dependency-resolution-modes) for more details.
